### PR TITLE
Adding button to download dataset as CSV

### PR DESCRIPTION
**Pull request by bazzyadb**
### Description
- Adding button to directly download dataset as CSV
- Only showing button once user selects a dataset
- Using streamlit download button to download the dataset
- Showing success prompt once user clicks on download button

- Relevant Issues : [Issue 171](https://github.com/intuit/Trapheus/issues/171)
- Type of change :
  - [x] New feature
  - [ ] Bug fix for existing feature
  - [ ] Code quality improvement
  - [ ] Addition or Improvement of tests
  - [ ] Addition or Improvement of documentation

### CODE REVIEW

File: labs/TrapheusAI/trapheusai_app.py [hunk 1]
```diff
@@ -98,6 +101,9 @@ def style_page():
             """
     streamlit.markdown(style, unsafe_allow_html=True)
 
+def showDownloadMessage():
+    streamlit.success("Downloading dataset to system")
```
Comment (stationeros) [inline labs/TrapheusAI/trapheusai_app.py:105]: Can we put the message as "Your dataset has been saved"

File: labs/TrapheusAI/trapheusai_app.py [hunk 2]
```diff
@@ -47,6 +47,9 @@ def handle_dataset_search():
                 repository_url = data["fullurl"]
                 df = pd.read_csv(repository_url)
                 streamlit.data_editor(df)
+                streamlit.download_button(
```
Comment (stationeros) [inline labs/TrapheusAI/trapheusai_app.py:50]: Can we check if we edit the dataset and then download are the changes retained?

File: labs/TrapheusAI/trapheusai_app.py (hunk 1 above)
Comment (bazzyadb) [reply labs/TrapheusAI/trapheusai_app.py]: Updated message

File: labs/TrapheusAI/trapheusai_app.py (hunk 2 above)
Comment (bazzyadb) [reply labs/TrapheusAI/trapheusai_app.py:50]: It won't retain the changes but we can modify this to retain the changes if required.

Comment (stationeros) [reply labs/TrapheusAI/trapheusai_app.py:50]: Merging this, can you open a new issue and work on saving the in memory CSV file

**Review by stationeros — APPROVED** [review]

**Comment by stationeros** [comment]
@bazzyadb  Please star the repo for showing us some support as well

**Comment by bazzyadb** [comment]
@stationeros @namitad Can one of you please add `hacktoberfest` and `hacktoberfest-accepted` to this pull request?

**Pull request merged**